### ISSUE TITLE
QMK firmware for Sparrow62 keyboard with RPI Pico controller on board.

### ISSUE THREAD

**Issue by azhizhinov** [74th/sparrow62-buildguide#3]
Hello.
There is no QMK firmware for Sparrow62 keyboard with RPI Pico controller on board, right?
So, there is no possibility to use VIA or VIAL GUIs to configure keyboard's layout?
Only KMK firmware available?
Thanks in advance.

**Comment by 74th** [comment]
Thank you for your message.
I had not planned to support QMK with that at this time, but if there is a request, I will try to support it.

**Comment by azhizhinov** [comment]
Thanks for you answer.
It will be great, if you'll find some time for this task.
I'm trying to figuring out the code by myself, but I've almost 0 knowledge.
I think code from Ferris keyboard can be used, but with STM32 -> RP2040 pins translation.

**Comment by azhizhinov** [comment]
Hello.
Do you have some news about QMK firmware for the Sparrow62 v2 with RPI Pico controller?

**Comment by azhizhinov** [comment]
Hello.
So, you can't help with the QMK firmware for the Sparrow 62 v2? I can't use the KMK firmware for some reasons, it's just doesn't work.

**Comment by 74th** [comment]
Sorry for the delay.
I'm getting serious about implementation this week.

WIP https://github.com/74th/qmk_firmware_sparrow_keyboard/blob/sparrow/keyboards/sparrow62/rev2/matrix.c

**Comment by 74th** [comment]
@azhizhinov 
Sorry for the wait.
We have added support for QMK Firmware and VIA (REMAP).
We would be happy if you could try using them if you like.

https://github.com/74th/sparrow62-buildguide/blob/master/sparrow62_v2.md#using-qmk-firmware--qmk-firmware%E3%81%AE%E4%BD%BF%E7%94%A8%E3%81%AB%E3%81%A4%E3%81%84%E3%81%A6

https://github.com/74th/qmk_firmware_sparrow_keyboard/tree/sparrow/keyboards/sparrow62/rev2

https://github.com/74th/qmk_firmware_sparrow_keyboard/wiki

**Comment by azhizhinov** [comment]
Hello.
Thank you a lot.
It's working. At first, I've checked only with the left half and no output from the keyboard had been received. In the QMK console were string with tries to connect to the MCP23017 and I've assumed the right half should be connected.
I have to confess, I've recreated your keyboard from scratch. :-)
Here the [KiCAD](https://github.com/user-attachments/files/16646052/20240817.AZHIZHINOV.SPROW63.00.zip) project and [models](https://github.com/user-attachments/files/16646063/20240817.AZHIZHINOV.SPROW63.00.zip) for FDM printers.
By the way, is there any possibility to change the firmware to work only with the left half connected? If so, that will be great.
Thank you and good luck.
